File: tree_primer/tree_primer__heap_judge.py
# 実行時間 10ms
def main():
    n, r = list(map(int, input().split()))

    for _ in [0] * (n - 1):
        a, b = list(map(int, input().split()))

        if a < b:
            return "NO"
    return "YES"

print(main())

File: tree_primer/tree_primer__heap_judge.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int[] n = ReadLine().Split().Select( int.Parse ).ToArray();

        foreach( int _ in Range( 0, n[0] - 1 ) ) {
            int[] a = ReadLine().Split().Select( int.Parse ).ToArray();

            if( a[0] < a[1] ) {
                WriteLine( "NO" );
                return;
            }
        }
        WriteLine( "YES" );
        return;
    }
}
